src/hardware/serialport/softmodem.cpp

```diff
@@ -355,6 +347,37 @@ char CSerialModem::GetChar(char * & scan) const {
 	return ch;
 }
 
+void CSerialModem::SetModemSpeed(uint32_t cfg_val) {
+	modem_bps_config   = cfg_val;
+	uint32_t reported_bps = UpdateConnectString();
+
+	if (modem_bps_config) {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will report connection speed "
+		        "of %d bits per second",
+		        GetPortNumber(),
+		        reported_bps);
+	} else {
+		LOG_MSG("SERIAL: Port %" PRIu8 " modem will synchronize reported speed "
+		        "to port baud rate",
+		        GetPortNumber());
+	}
+}
+
+uint32_t CSerialModem::UpdateConnectString() {
+	constexpr auto min_bps = 300u;
+	constexpr auto max_bps = 115200u;
+	uint32_t connect_val;
+	if (modem_bps_config != 0) {
+		connect_val = clamp(modem_bps_config, min_bps, max_bps);
```

Comment:
Question: shouldn't we log a warning about using out of bounds rates and informing the user about what value we ended up using?